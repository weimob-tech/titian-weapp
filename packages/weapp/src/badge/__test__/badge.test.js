const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('badge');
    comp.triggerLifeTime('attached');
    expect(comp).toMatchSnapshot();
  });

  it('should render success with content', () => {
    const comp = getComponent('badge', {
      content: '1'
    });
    expect(comp).toMatchSnapshot();
  });

  it('should render success with icon', () => {
    const comp = getComponent('badge', {
      icon: 'home'
    });
    expect(comp).toMatchSnapshot();
  });
  it('should render success with dot', () => {
    const comp = getComponent('badge', {
      dot: true
    });
    expect(comp).toMatchSnapshot();
  });
});
