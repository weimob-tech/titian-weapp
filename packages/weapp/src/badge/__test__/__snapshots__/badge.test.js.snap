// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="titian-badge"
  >
    <wx-view
      class="ext-class titian-badge-container-hidden titian-badge-container"
      style=""
    />
  </wx-view>
</main>
`;

exports[`test component should render success with content 1`] = `
<main>
  <wx-view
    class="titian-badge"
  >
    <wx-view
      class="ext-class titian-badge-container-circle titian-badge-container"
      style=""
    >
      1
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render success with dot 1`] = `
<main>
  <wx-view
    class="titian-badge"
  >
    <wx-view
      class="ext-class titian-badge-container-dot titian-badge-container"
      style=""
    />
  </wx-view>
</main>
`;

exports[`test component should render success with icon 1`] = `
<main>
  <wx-view
    class="titian-badge"
  >
    <wx-view
      class="ext-class titian-badge-container-circle titian-badge-container"
      style=""
    >
      <ti-icon>
        <wx-view
          class="titian-icon-base titian-icon titian-icon-home ext-class"
          style="font-size:20rpx"
          bind:tap="onClick"
        />
      </ti-icon>
    </wx-view>
  </wx-view>
</main>
`;
